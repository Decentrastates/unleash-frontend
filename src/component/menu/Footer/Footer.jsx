import React from 'react';

import { List, ListItem, ListItemText, Grid } from '@material-ui/core';

import ShowApiDetailsContainer from '../../api/show-api-details-container';

import { useStyles } from './Footer.styles';

export const Footer = () => {
    const styles = useStyles();

    return (
        <footer className={styles.footer}>
            <Grid container>
                <Grid item xs={3}>
                    <section title="Client SDKs">
                        <h4>Client SDKs</h4>
                        <List className={styles.list}>
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://github.com/Unleash/unleash-client-node"
                                            className={styles.link}
                                        >
                                            Node.js
                                        </a>
                                    }
                                />
                            </ListItem>
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://github.com/Unleash/unleash-client-java"
                                            className={styles.link}
                                        >
                                            Java
                                        </a>
                                    }
                                />
                            </ListItem>
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://github.com/Unleash/unleash-client-go"
                                            className={styles.link}
                                        >
                                            Go
                                        </a>
                                    }
                                />
                            </ListItem>{' '}
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://github.com/Unleash/unleash-client-ruby"
                                            className={styles.link}
                                        >
                                            Ruby
                                        </a>
                                    }
                                />
                            </ListItem>{' '}
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://github.com/Unleash/unleash-client-python"
                                            className={styles.link}
                                        >
                                            Python
                                        </a>
                                    }
                                />
                            </ListItem>
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://github.com/Unleash/unleash-client-core"
                                            className={styles.link}
                                        >
                                            .Net Core
                                        </a>
                                    }
                                />
                            </ListItem>
                            <ListItem className={styles.listItem}>
                                <ListItemText
                                    primary={
                                        <a
                                            href="https://unleash.github.io/docs/client_sdk"
                                            className={styles.link}
                                        >
                                            All client SDKs
                                        </a>
                                    }
                                />
                            </ListItem>
                        </List>
                    </section>
                </Grid>
                <Grid item xs={12}>
                    <ShowApiDetailsContainer />
                </Grid>
            </Grid>
        </footer>
    );
};

export default Footer;
