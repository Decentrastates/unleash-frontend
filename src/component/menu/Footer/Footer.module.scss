.list {
    padding: 0;
    margin: 0;
}

.listItem {
    padding: 0;
    margin: 0;
}

.listItem a {
    text-decoration: none;
    color: #000;
}

.footer {
    background: #fff;
    padding: 2rem 4rem;
    color: #000;
    width: 100%;
}
