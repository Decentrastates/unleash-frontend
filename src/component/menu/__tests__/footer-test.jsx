import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import Footer from '../Footer/Footer';
import theme from '../../../themes/main-theme';

const mockStore = {
    uiConfig: {
        toJS: () => ({
            flags: {
                P: true,
            },
        }),
    },
};

const mockReducer = state => state;

test('should render DrawerMenu', () => {
    const tree = renderer.create(
        <Provider store={createStore(mockReducer, mockStore)}>
            <ThemeProvider theme={theme}>
                <MemoryRouter>
                    <Footer />
                </MemoryRouter>
            </ThemeProvider>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
});

test('should render DrawerMenu with "features" selected', () => {
    const tree = renderer.create(
        <Provider store={createStore(mockReducer, mockStore)}>
            <ThemeProvider theme={theme}>
                <MemoryRouter initialEntries={['/features']}>
                    <Footer />
                </MemoryRouter>
            </ThemeProvider>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
});
