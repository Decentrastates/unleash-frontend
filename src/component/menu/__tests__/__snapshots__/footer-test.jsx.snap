// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DrawerMenu 1`] = `
<footer
  className="makeStyles-footer-1"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <section
        title="Client SDKs"
      >
        <h4>
          Client SDKs
        </h4>
        <ul
          className="MuiList-root makeStyles-list-2 MuiList-padding"
        >
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-node"
                >
                  Node.js
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-java"
                >
                  Java
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-go"
                >
                  Go
                </a>
              </span>
            </div>
          </li>
           
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-ruby"
                >
                  Ruby
                </a>
              </span>
            </div>
          </li>
           
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-python"
                >
                  Python
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-core"
                >
                  .Net Core
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://unleash.github.io/docs/client_sdk"
                >
                  All client SDKs
                </a>
              </span>
            </div>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <section
        title="API details"
      >
        <h4>
          undefined undefined
        </h4>
        <br />
        <br />
        <br />
        <small />
        <br />
      </section>
    </div>
  </div>
</footer>
`;

exports[`should render DrawerMenu with "features" selected 1`] = `
<footer
  className="makeStyles-footer-1"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <section
        title="Client SDKs"
      >
        <h4>
          Client SDKs
        </h4>
        <ul
          className="MuiList-root makeStyles-list-2 MuiList-padding"
        >
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-node"
                >
                  Node.js
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-java"
                >
                  Java
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-go"
                >
                  Go
                </a>
              </span>
            </div>
          </li>
           
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-ruby"
                >
                  Ruby
                </a>
              </span>
            </div>
          </li>
           
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-python"
                >
                  Python
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://github.com/Unleash/unleash-client-core"
                >
                  .Net Core
                </a>
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters"
            disabled={false}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <a
                  href="https://unleash.github.io/docs/client_sdk"
                >
                  All client SDKs
                </a>
              </span>
            </div>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <section
        title="API details"
      >
        <h4>
          undefined undefined
        </h4>
        <br />
        <br />
        <br />
        <small />
        <br />
      </section>
    </div>
  </div>
</footer>
`;
